lib/fhir_client/client.rb

```diff
@@ -289,12 +273,19 @@ def try_conformance_formats(default_format)
         rescue
           @cached_capability_statement = nil
         end
-        unless @cached_capability_statement
+        if @cached_capability_statement.nil? || !@cached_capability_statement.fhirVersion.starts_with?('4')
```

Comment:
You never want to cache R4 capability statements? Why just R4?

Reply:
I think I understand your question...

This change was to differentiate between STU3 and R4 CapabilityStatements, which otherwise could be created into a `FHIR::Model` or `FHIR::STU3` model depending on the configuration of the client.

The unless statement was changed because it was no longer sufficient to check if the CapabilityStatement could be parsed by FHIR::CapabilityStatement alone.  Luckily, fhirVersion is required in the CapabilityStatement so we now check that it was parsed (if it wasn't it's probably a DSTU2 ConformanceStatement) and that the version starts with '4' to indicate R4.  If it is then we cache it and break the loop, otherwise we try the others like we used to.

The CapabilityStatements are still cached, but I'll update the autodetect tests to be sure.

I didn't break up unit test that exemplified this from the fix, but if you swap `try_conformance_format` for the old implementation you'll get a failure like this if you run `test_autodetect_stu3`
```
Failure: test_autodetect_stu3(MultiversionTest):
  Expected Version to be stu3, but found r4.
  <false> is not true.
```

because the STU3 CapabilityStatement was parsed into an R4 `FHIR::Model`.

Let me know if I totally misunderstood...

[fhirVersion in CapabilityStatement](https://www.hl7.org/fhir/capabilitystatement-definitions.html#CapabilityStatement.fhirVersion)

Reply:
Honestly, your explanation didn't make a whole lot of sense to me, so I went back and read the code for that method from beginning to end. I think it is OK, except I suspect the `use_dstu2` is in the wrong place.

Reply:
Sorry for the poor explanation.  And yes I agree the `use_dstu2` should be moved.